Pipeline/SimplePackageDeploy/SimplePackageDeploy.groovy

```diff
@@ -0,0 +1,287 @@
+ *
+ * INTERNALS: 
+ *
+ *  1) Reads the options --workDir, --tarFileName, and --hlq from the command line. Pass the optional 
+ *     parm --packageWithExtension if the build package is created with addExtension option. 
+ *  2) Reads the options copyModeMap and targetLibLLQMap from the SimplePackageDeploy.properties 
+ *     file.
+ *  3) Creates a temporary package extract folder in the path provided using the 
+ *     --workDir option.
+ *  4) Untars the deployment package specified by --tarFileName to the temporary 
+ *     folder created in the previous step.
+ *  5) Reads the BuildReport.json file from the temporary folder.
+ *  6) Identifies all the components to be deployed from the BuildReport.json file.
+ *  7) Identifies deploy types for the components to be deployed from the 
+ *     BuildReport.json file.
+ *  8) Identifies the target libraries based on the deploy type by concatenating the
+ *     --hlq value with the correspoding LLQ value from the JSON string - targetLibLLQMap.
+ *  9) Identifies the copy mode based on the deploy type and corresponding copy mode 
+ *     from the JSON string - copyModeMap.
+ * 10) Copies the components using the copy mode to the respective target libraries.
+ * 11) Deletes the temporary package extract folder after deployment.
+ *
+ * RETURNS:
+ *    Exit Status 0       -  Success
+ *    Exit Status 1       -  Failure
+ *
+ ************************************************************************************/
+
+@Field Properties props = null
+props = parseInput(args)
+
+def startTime = new Date()
+props.startTime = startTime.format("yyyyMMdd.hhmmss.mmm")
+println("** SimplePackageDeploy start at $props.startTime")
+
+// Map of last level dataset qualifier to DBB CopyToPDS CopyMode.
+def copyModeMap = parseJSONStringToMap(props.copyModeMap)
+// Map of last level dataset qualifier to Target PDS last level qualifier.
+def targetLibLLQMap = parseJSONStringToMap(props.targetLibLLQMap)
+
+// Define and create tar file extract directory 
+def tarExtractDirName = "${props.workDir}/DeployFiles_${props.startTime}"
+def tarExtractDir = new File("${tarExtractDirName}")
+
+!tarExtractDir.exists() ?: tarExtractDir.deleteDir()
+tarExtractDir.mkdirs()
+println("** Created tar file extract directory $tarExtractDirName")       
+
+// Check if the tar file provided is existing 
+def tarFile = new File("${props.tarFileName}")
+
+if(!tarFile.exists()){
+    println("** Tar file not found at $tarFile")
+    System.exit(1)
+}
+
+
+// Untar the tar file
+println("** Untar file at $tarFile.")
+
+def processCmd = ["sh", "-c", "tar -C $tarExtractDirName -xvf $tarFile"]
+
+def rc = runProcess(processCmd)
+if (rc == 0) println("Package untar done to $tarExtractDirName")
+else {
+	println("Failed to untar $tarFile")
+	System.exit(1)
+}
+
+
+// Read build report data
+def listDirFiles = new File("$tarExtractDirName")
+def buildReportFlag = 0
+
+// Find the BuildReport.json file for deployment 
+listDirFiles.eachFile {
```

Comment:
Rather than searching for the BuildReport.json in the temp directory, I think we could just assume it to be on the root level for now - this is how the packaging is performed and you make the reference to it.

The logic around a cumulative package of multiple build reports is based on the `buildReportOrder.txt`, but this is fairly complex, while contents of the build reports can overlap. For the first release of this sample script, I would propose, if the buildReportOrder file is found, the script should stop processing and write out to the log that it is not supporting this packaging layout. Further on that, to me this is rather an enhancement of the packaging script to build a cumulative build report as part of its processing, than rebuilding a cumulative process here!

Reply:
Updated the script and readme file for single BuildReport.json as discussed.